waiter/integration/waiter/token_request_test.clj

```diff
@@ -1366,6 +1366,77 @@
         (finally
           (delete-token-and-assert waiter-url token))))))
 
+(deftest ^:parallel ^:integration-fast test-maintenance-mode
+  (testing-using-waiter-url
+    (let [service-name (rand-name)
+          token (create-token-name waiter-url service-name)
+          current-user (retrieve-username)
+          service-description (-> (kitchen-request-headers :prefix "")
+                                  (assoc :name service-name
+                                         :permitted-user "*"
+                                         :run-as-user current-user))
```

Comment:
Use `https-redirect true`, `interstitial-secs 60`, and `run-as-user *` and to trigger consent handler, interstitial and https redirect and make sure they are not triggered in the test.